src/RESTController.js

```diff
@@ -159,27 +159,28 @@ const RESTController = {
         headers[key] = customHeaders[key];
       }
 
-      function handleProgress(event) {
+      function handleProgress(type, event) {
         if (options && typeof options.progress === 'function') {
           if (event.lengthComputable) {
             options.progress(
```

Comment:
I pushed it at the beginning, or

```js
progress(event.loaded / event.total, {
  type,
  event
});
```

@dplewis , @ladigitale let me know your thoughts

Reply:
I do think there is. I was reluctant to just add it and the end, because if somehow, why not just enclose it in object to allow future values instead of
```js
progress(progressValue, loaded, total, type, ...restArgs)
```

Reply:
it would be perfect.
In fact when we've got the original event i think we've got the type.
The targets of each event have a different type who gives the needed imformation.

Reply:
Is that so? Have you tried it? haven't looked into values of `event` tho.

Reply:
I just logged the target propety one is XMLHttpRequest, and the other is XMLHttpRequestUpload

Reply:
Feel free to make suggestion on the PR using the suggestion feature of Github. :smile: